### DISCUSSION TITLE
short code for font awesome

### DISCUSSION THREAD

**Discussion by kreincke** [bootscore/bootscore#376] [Ideas]
Thankfully, bootScore brings Font Awesome right inside. This reduces traffic and imponderables in terms of data transfer to third parties. Therefore, bootScore can legitimately say that it has installed Font Awesome "in a GDPR-friendly way".

The icons are supposed to be used by typing HTML code like '&lt;i class="fa-regular fa-square-check"&gt;&lt;/i&gt;' into blocks at Gutenberg level. Unfortunately, this occasionally destroys the selected block type and its content (e.g. lists) - which is not a bug of bootScore, but a weakness of WordPress: HTML-level editing is dangerous in general.

But bootScore could support its users by defining and activating a short code in the general functions.php:

```
function insert_bsfa($atts){
  $atts = (array) $atts;
  $vstr = "";
  foreach ( $atts as $value ) {
    $vstr = $vstr . " $value";
  }
  return '<i class="' . $vstr . '"></i>';
}
add_shortcode('bsfa', 'insert_bsfa');
```
The example above is reduced to `[bsfa fa-regular fa-square-check]` into a paragraph or list item block.

The advantage is, that this handling allows the same functionality as the original code based integration, but prevents us from occasional accidents on HTML-Editor level.

If you agree with that, I could create a respective pull request.

br kr

**Comment by crftwrk** [comment]
I think this is a great idea! Yes, please PR ;-)
